gcloud-java-compute/src/main/java/com/google/gcloud/compute/DiskInfo.java

```diff
@@ -0,0 +1,439 @@
+  }
+
+  /**
+   * Returns the unique identifier for the disk; defined by the service.
+   */
+  public String id() {
+    return id;
+  }
+
+  /**
+   * Returns the disk configuration.
+   */
+  @SuppressWarnings("unchecked")
+  public <T extends DiskConfiguration> T configuration() {
+    return (T) configuration;
+  }
+
+  /**
+   * Returns the disk identity.
+   */
+  public DiskId diskId() {
+    return diskId;
+  }
+
+  /**
+   * Returns the creation status of the disk.
+   */
+  public CreationStatus creationStatus() {
+    return creationStatus;
+  }
+
+  /**
+   * Returns a textual description of the disk.
+   */
+  public String description() {
+    return description;
+  }
+
+  /**
+   * Returns all applicable publicly visible licenses for the disk.
+   */
+  public List<LicenseId> licenses() {
+    return licenses;
+  }
+
+  /**
+   * Returns all the identities of the instances this disk is attached to.
+   */
+  public List<InstanceId> attachedInstances() {
+    return attachedInstances;
+  }
+
+  /**
+   * Returns the last attach timestamp in milliseconds since epoch.
+   */
+  public Long lastAttachTimestamp() {
+    return lastAttachTimestamp;
+  }
+
+  /**
+   * Returns the last detach timestamp in milliseconds since epoch.
+   */
+  public Long lastDetachTimestamp() {
+    return lastDetachTimestamp;
+  }
+
+  /**
+   * Returns a builder for the object.
+   */
+  public Builder toBuilder() {
+    return new BuilderImpl(this);
+  }
+
+  @Override
+  public String toString() {
+    return MoreObjects.toStringHelper(this)
+        .add("id", id)
+        .add("diskId", diskId)
+        .add("configuration", configuration)
+        .add("creationTimestamp", creationTimestamp)
+        .add("creationStatus", creationStatus)
+        .add("description", description)
+        .add("licenses", licenses)
+        .add("attachedInstances", attachedInstances)
+        .add("lastAttachTimestamp", lastAttachTimestamp)
+        .add("lastDetachTimestamp", lastDetachTimestamp)
+        .toString();
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(diskId, configuration, creationTimestamp, creationStatus, description,
+        licenses, attachedInstances, lastAttachTimestamp, lastDetachTimestamp);
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    return obj != null
+        && obj.getClass().equals(DiskInfo.class)
+        && Objects.equals(toPb(), ((DiskInfo) obj).toPb());
+  }
+
+  /**
+   * Returns a builder for a {@code DiskInfo} object given its identity and configuration. Use
+   * {@link StandardDiskConfiguration} to create a simple disk given its type and size. Use
+   * {@link SnapshotDiskConfiguration} to create a disk from a snapshot. Use
+   * {@link ImageDiskConfiguration} to create a disk from a disk image.
+   */
+  public static Builder builder(DiskId diskId, DiskConfiguration configuration) {
+    return new BuilderImpl().diskId(diskId).configuration(configuration);
```

Comment:
Any reason not to move these arguments to the empty constructor?
Also, if we do so, the checkNotNull(configuration) in the DiskInfo constructor may not be necessary.